packages/server/src/app/services/auth/auth.service.ts

```diff
@@ -0,0 +1,94 @@
+import {
+    Injectable,
+    Logger,
+    UnauthorizedException,
+    InternalServerErrorException,
+    NotFoundException,
+} from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { verify, argon2id } from 'argon2';
+import { nanoid } from 'nanoid';
+
+import { User, FrontendUser, CreateUser, AuditSession } from '@dragonfish/models/users';
+import { UsersStore } from '../../db/users/users.store';
+import { JwtPayload } from '@dragonfish/models/auth';
+import { IAuth } from '../../shared/auth';
+
+@Injectable()
+export class AuthService implements IAuth {
+    private readonly logger: Logger = new Logger(AuthService.name);
+
+    constructor(private readonly usersService: UsersStore, private readonly jwtService: JwtService) {}
+
+    async validateUser(email: string, password: string): Promise<User> {
+        const potentialUser = await this.usersService.findOneByEmail(email);
+        if (potentialUser) {
+            try {
+                if (await verify(potentialUser.password, password, { type: argon2id })) {
+                    return potentialUser;
+                } else {
+                    throw new UnauthorizedException('Either your email or password is invalid.');
+                }
+            } catch (err) {
+                throw new InternalServerErrorException(
+                    'Something went wrong logging you in. Try again in a little bit.',
+                );
+            }
+        } else {
+            throw new NotFoundException("Looks like you don't exist yet. Why not try signing up?");
```

Comment:
You were asking about error logging--places like this should all be logged (unless we get exception logging by default, in which case that's good enough. Unless there's more data we'd like to record about the error that an exception alone won't capture)

Reply:
let's go over that in a future PR